core/src/main/java/org/jdbi/v3/core/rewriter/InternalRewrittenStatement.java

```diff
@@ -36,49 +37,55 @@
 
     @Override
     public void bind(Binding params, PreparedStatement statement) throws SQLException {
-        if (stmt.positionalOnly) {
-            // no named params, is easy
-            boolean finished = false;
-            for (int i = 0; !finished; ++i) {
-                final Optional<Argument> a = params.findForPosition(i);
-                if (a.isPresent()) {
-                    try {
-                        a.get().apply(i + 1, statement, this.context);
-                    } catch (SQLException e) {
-                        throw new UnableToExecuteStatementException(
-                                String.format("Exception while binding positional param at (0 based) position %d",
-                                        i), e, context);
-                    }
-                } else {
-                    finished = true;
-                }
-            }
+        if (stmt.isPositional()) {
+            bindPositional(params, statement);
         } else {
-            //List<String> named_params = stmt.params;
-            int i = 0;
-            for (String named_param : stmt.params) {
-                if ("*".equals(named_param)) {
-                    continue;
-                }
-                final int index = i;
-                Argument a = findFirstPresent(
-                        () -> params.findForName(named_param),
-                        () -> params.findForPosition(index))
-                        .orElseThrow(() -> {
-                            String msg = String.format("Unable to execute, no named parameter matches " +
-                                            "\"%s\" and no positional param for place %d (which is %d in " +
-                                            "the JDBC 'start at 1' scheme) has been set.",
-                                    named_param, index, index + 1);
-                            return new UnableToExecuteStatementException(msg, context);
-                        });
+            bindNamed(params, statement);
+        }
+    }
+
+    private void bindPositional(Binding params, PreparedStatement statement) {
+        int paramCount = stmt.getParams().size();
+        for (int i = 0; i < paramCount; i++) {
+            int index = i;
+            Argument a = params.findForPosition(i)
```

Comment:
I think we could avoid declaring the argument variable and just invoke `apply` after `orElseThrow`.